### PR TITLE
S3 us17 login

### PR DESCRIPTION
**Pull request by DESCARAVAGE**

### CODE REVIEW

File: src/Controller/LoginController.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+namespace App\Controller;
+
+use App\Model\UserManager;
+
+class LoginController extends AbstractController
+{
+    public function login(): string
+    {
+        $errors = [];
+        $connexion = [];
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:12]: can-you-speak-english-sonic

Comment (ThibaultFoure) [reply src/Controller/LoginController.php:12]: et tu peux faire de manière plus concise `$errors = $connection = [];`

File: src/Controller/LoginController.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        if ($_SERVER["REQUEST_METHOD"] === "POST") {
+            $connexion = array_map('trim', $_POST);
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:15]: right-here-dwayne-johnson

File: src/Controller/LoginController.php [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            $errors = $this->validate($connexion);
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:16]: right-here-ignace-aleya

File: src/Controller/LoginController.php [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    public function logout()
+    {
+        if (!empty($_SESSION['user'])) {
+            unset($_SESSION['user']);
+        }
+
+        header('Location: /');
+    }
+
+    private function validate(array $connexion): array
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:32]: over-here-here

File: src/Controller/LoginController.php [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        $errors = [];
+        if (empty($connexion['email'])) {
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:35]: right-here-neil-degrasse-tyson

File: src/Controller/LoginController.php [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            $errors[] = 'Le champ email ne doit pas être vide';
+        }
+        if (empty($connexion['password'])) {
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:38]: right-here-forrest-starling

File: src/Controller/LoginController.php [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            $errors[] = 'Le mot de passe ne doit pas être vide';
+        }
+        if (!filter_var($connexion['email'], FILTER_VALIDATE_EMAIL)) {
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:41]: right-here-echo-gaming

File: src/Controller/LoginController.php [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            $errors[] = "format d'email invalide";
+        }
+        if (empty($errors)) {
+            $userManager = new UserManager();
+            $user = $userManager->selectOneByEmail($connexion['email']);
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:46]: look-you-see

File: src/Controller/LoginController.php [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+            if ($user) {
+                if (password_verify($connexion['password'], $user['password'])) {
```
Comment (ThibaultFoure) [inline src/Controller/LoginController.php:48]: over-there-john-finch

File: src/View/Login/login.html.twig [hunk 10]
```diff
@@ -0,0 +1,29 @@
+{% extends 'layout.html.twig' %}
+{% block stylesheet %}
+	<link rel="stylesheet" href="/assets/css/login.css">
+{% endblock %}
+{% block content %}
+	<div class="container">
+		<form method="POST" action="" novalidate>
```
Comment (ThibaultFoure) [inline src/View/Login/login.html.twig:7]: novalidate à supprimer

**Review by ThibaultFoure — CHANGES_REQUESTED** [review]

**Review by JSDerouet — APPROVED** [review]

**Review by ThibaultFoure — APPROVED** [review]

**Review by ThibaultFoure — APPROVED** [review]

**Pull request merged**